In [55]:
import pandas as pd
import numpy as np
from sklearn import preprocessing
import matplotlib.pyplot as plt 
plt.rc("font", size=14)
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score
from sklearn.metrics import classification_report
from sklearn.metrics import confusion_matrix
import seaborn as sns
sns.set(style="white")
sns.set(style="whitegrid", color_codes=True)



i started importing necessary libraries, including pandas, numpy,scikit-lear, and matplotlib.you also set some formating
for my plots using'plt.rc.

In [23]:
df = pd.read_csv(r'C:\Users\Dell\Downloads\Global Power Plant Database.csv')
df


country country_long                          name   gppd_idnr  \
0       IND        India              ACME Solar Tower  WRI1020239   
1       IND        India           ADITYA CEMENT WORKS  WRI1019881   
2       IND        India      AES Saurashtra Windfarms  WRI1026669   
3       IND        India                   AGARTALA GT  IND0000001   
4       IND        India                  AKALTARA TPP  IND0000002   
..      ...          ...                           ...         ...   
902     IND        India                  YERMARUS TPP  IND0000513   
903     IND        India  Yelesandra Solar Power Plant  WRI1026222   
904     IND        India  Yelisirur wind power project  WRI1026776   
905     IND        India                   ZAWAR MINES  WRI1019901   
906     IND        India       iEnergy Theni Wind Farm  WRI1026761   

     capacity_mw  latitude  longitude primary_fuel other_fuel1 other_fuel2  \
0            2.5   28.1839    73.2407        Solar         NaN         NaN   
1           98.0   24.7663    74.6090         Coal         NaN         NaN   
2           39.2   21.9038    69.3732         Wind         NaN         NaN   
3          135.0   23.8712    91.3602          Gas         NaN         NaN   
4         1800.0   21.9603    82.4091         Coal         Oil         NaN   
..           ...       ...        ...          ...         ...         ...   
902       1600.0   16.2949    77.3568         Coal         Oil         NaN   
903          3.0   12.8932    78.1654        Solar         NaN         NaN   
904         25.5   15.2758    75.5811         Wind         NaN         NaN   
905         80.0   24.3500    73.7477         Coal         NaN         NaN   
906         16.5    9.9344    77.4768         Wind         NaN         NaN   

     ...  year_of_capacity_data  generation_gwh_2013 generation_gwh_2014  \
0    ...                    NaN                  NaN                 NaN   
1    ...                    NaN                  NaN                 NaN   
2    ...                    NaN                  NaN                 NaN   
3    ...                 2019.0                  NaN          617.789264   
4    ...                 2019.0                  NaN         3035.550000   
..   ...                    ...                  ...                 ...   
902  ...                 2019.0                  NaN                 NaN   
903  ...                    NaN                  NaN                 NaN   
904  ...                    NaN                  NaN                 NaN   
905  ...                    NaN                  NaN                 NaN   
906  ...                    NaN                  NaN                 NaN   

    generation_gwh_2015 generation_gwh_2016 generation_gwh_2017  \
0                   NaN                 NaN                 NaN   
1                   NaN                 NaN                 NaN   
2                   NaN                 NaN                 NaN   
3            843.747000          886.004428          663.774500   
4           5916.370000         6243.000000         5385.579736   
..                  ...                 ...                 ...   
902            0.994875          233.596650          865.400000   
903                 NaN                 NaN                 NaN   
904                 NaN                 NaN                 NaN   
905                 NaN                 NaN                 NaN   
906                 NaN                 NaN                 NaN   

     generation_gwh_2018  generation_gwh_2019         generation_data_source  \
0                    NaN                  NaN                            NaN   
1                    NaN                  NaN                            NaN   
2                    NaN                  NaN                            NaN   
3             626.239128                  NaN  Central Electricity Authority   
4            7279.000000                  NaN  Central Electricity Authority   
..                   ...    

In [24]:
df.isna().sum()

country                       0
country_long                  0
name                          0
gppd_idnr                     0
capacity_mw                   0
latitude                     46
longitude                    46
primary_fuel                  0
other_fuel1                 709
other_fuel2                 906
other_fuel3                 907
commissioning_year          380
owner                       565
source                        0
url                           0
geolocation_source           19
wepp_id                     907
year_of_capacity_data       388
generation_gwh_2013         907
generation_gwh_2014         509
generation_gwh_2015         485
generation_gwh_2016         473
generation_gwh_2017         467
generation_gwh_2018         459
generation_gwh_2019         907
generation_data_source      458
estimated_generation_gwh    907
dtype: int64

In The above i tried to check the missing data in my dataset using 'df.isna().sum()

In [25]:
df.fillna(df.mean(), inplace=True)

C:\Users\Dell\AppData\Local\Temp\ipykernel_10184\820435583.py:1: FutureWarning: The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.fillna(df.mean(), inplace=True)


In [26]:
df.isna().sum()

country                       0
country_long                  0
name                          0
gppd_idnr                     0
capacity_mw                   0
latitude                      0
longitude                     0
primary_fuel                  0
other_fuel1                 709
other_fuel2                 906
other_fuel3                 907
commissioning_year            0
owner                       565
source                        0
url                           0
geolocation_source           19
wepp_id                     907
year_of_capacity_data         0
generation_gwh_2013         907
generation_gwh_2014           0
generation_gwh_2015           0
generation_gwh_2016           0
generation_gwh_2017           0
generation_gwh_2018           0
generation_gwh_2019         907
generation_data_source      458
estimated_generation_gwh    907
dtype: int64

In [27]:
column_names = df.columns
for column in column_names:
    print(column + ' - ' + str(df[column].isnull().sum()))

country - 0
country_long - 0
name - 0
gppd_idnr - 0
capacity_mw - 0
latitude - 0
longitude - 0
primary_fuel - 0
other_fuel1 - 709
other_fuel2 - 906
other_fuel3 - 907
commissioning_year - 0
owner - 565
source - 0
url - 0
geolocation_source - 19
wepp_id - 907
year_of_capacity_data - 0
generation_gwh_2013 - 907
generation_gwh_2014 - 0
generation_gwh_2015 - 0
generation_gwh_2016 - 0
generation_gwh_2017 - 0
generation_gwh_2018 - 0
generation_gwh_2019 - 907
generation_data_source - 458
estimated_generation_gwh - 907


In [28]:
columns_to_fill = ['primary_fuel','country','capacity_mw','country_long',"latitude","longitude",'other_fuel1','other_fuel2','other_fuel3','commissioning_year','owner','geolocation_source','wepp_id','year_of_capacity_data','generation_gwh_2013','generation_gwh_2014','generation_gwh_2015','generation_gwh_2016','generation_gwh_2017','generation_gwh_2018','generation_gwh_2019','generation_data_source','estimated_generation_gwh']
df = df[columns_to_fill].fillna(0)

In [29]:
df.isnull().sum()

primary_fuel                0
country                     0
capacity_mw                 0
country_long                0
latitude                    0
longitude                   0
other_fuel1                 0
other_fuel2                 0
other_fuel3                 0
commissioning_year          0
owner                       0
geolocation_source          0
wepp_id                     0
year_of_capacity_data       0
generation_gwh_2013         0
generation_gwh_2014         0
generation_gwh_2015         0
generation_gwh_2016         0
generation_gwh_2017         0
generation_gwh_2018         0
generation_gwh_2019         0
generation_data_source      0
estimated_generation_gwh    0
dtype: int64

In [30]:
df

primary_fuel country  capacity_mw country_long  latitude  longitude  \
0          Solar     IND          2.5        India   28.1839    73.2407   
1           Coal     IND         98.0        India   24.7663    74.6090   
2           Wind     IND         39.2        India   21.9038    69.3732   
3            Gas     IND        135.0        India   23.8712    91.3602   
4           Coal     IND       1800.0        India   21.9603    82.4091   
..           ...     ...          ...          ...       ...        ...   
902         Coal     IND       1600.0        India   16.2949    77.3568   
903        Solar     IND          3.0        India   12.8932    78.1654   
904         Wind     IND         25.5        India   15.2758    75.5811   
905         Coal     IND         80.0        India   24.3500    73.7477   
906         Wind     IND         16.5        India    9.9344    77.4768   

    other_fuel1 other_fuel2  other_fuel3  commissioning_year  ...  \
0             0           0          0.0         2011.000000  ...   
1             0           0          0.0         1997.091082  ...   
2             0           0          0.0         1997.091082  ...   
3             0           0          0.0         2004.000000  ...   
4           Oil           0          0.0         2015.000000  ...   
..          ...         ...          ...                 ...  ...   
902         Oil           0          0.0         2016.000000  ...   
903           0           0          0.0         1997.091082  ...   
904           0           0          0.0         1997.091082  ...   
905           0           0          0.0         1997.091082  ...   
906           0           0          0.0         1997.091082  ...   

    year_of_capacity_data generation_gwh_2013  generation_gwh_2014  \
0                  2019.0                 0.0          2431.823590   
1                  2019.0                 0.0          2431.823590   
2                  2019.0                 0.0          2431.823590   
3                  2019.0                 0.0           617.789264   
4                  2019.0                 0.0          3035.550000   
..                    ...                 ...                  ...   
902                2019.0                 0.0          2431.823590   
903                2019.0                 0.0          2431.823590   
904                2019.0                 0.0          2431.823590   
905                2019.0                 0.0          2431.823590   
906                2019.0                 0.0          2431.823590   

     generation_gwh_2015  generation_gwh_2016  generation_gwh_2017  \
0            2428.226946          2467.936859          2547.759305   
1            2428.226946          2467.936859          2547.759305   
2            2428.226946          2467.936859          2547.759305   
3             843.747000           886.004428           663.774500   
4            5916.370000          6243.000000          5385.579736   
..                   ...                  ...                  ...   
902             0.994875           233.596650           865.400000   
903          2428.226946          2467.936859          2547.759305   
904          2428.226946          2467.936859          2547.759305   
905          2428.226946          2467.936859          2547.759305   
906          2428.226946          2467.936859          2547.759305   

     generation_gwh_2018  generation_gwh_2019         generation_data_source  \
0            2600.804099                  0.0                              0   
1            2600.804099                  0.0                              0   
2            2600.804099                  0.0                              0   
3             626.239128                  0.0  Central Electricity Authority   
4            7279.000000                  0.0  Central Electricity Authority   
..                   ...                  ...                            ...   
902           686.500000        

In [31]:
columns_to_drop = ['owner',]
df = df.drop(columns=columns_to_drop, axis=1)
df

primary_fuel country  capacity_mw country_long  latitude  longitude  \
0          Solar     IND          2.5        India   28.1839    73.2407   
1           Coal     IND         98.0        India   24.7663    74.6090   
2           Wind     IND         39.2        India   21.9038    69.3732   
3            Gas     IND        135.0        India   23.8712    91.3602   
4           Coal     IND       1800.0        India   21.9603    82.4091   
..           ...     ...          ...          ...       ...        ...   
902         Coal     IND       1600.0        India   16.2949    77.3568   
903        Solar     IND          3.0        India   12.8932    78.1654   
904         Wind     IND         25.5        India   15.2758    75.5811   
905         Coal     IND         80.0        India   24.3500    73.7477   
906         Wind     IND         16.5        India    9.9344    77.4768   

    other_fuel1 other_fuel2  other_fuel3  commissioning_year  ...  \
0             0           0          0.0         2011.000000  ...   
1             0           0          0.0         1997.091082  ...   
2             0           0          0.0         1997.091082  ...   
3             0           0          0.0         2004.000000  ...   
4           Oil           0          0.0         2015.000000  ...   
..          ...         ...          ...                 ...  ...   
902         Oil           0          0.0         2016.000000  ...   
903           0           0          0.0         1997.091082  ...   
904           0           0          0.0         1997.091082  ...   
905           0           0          0.0         1997.091082  ...   
906           0           0          0.0         1997.091082  ...   

    year_of_capacity_data  generation_gwh_2013  generation_gwh_2014  \
0                  2019.0                  0.0          2431.823590   
1                  2019.0                  0.0          2431.823590   
2                  2019.0                  0.0          2431.823590   
3                  2019.0                  0.0           617.789264   
4                  2019.0                  0.0          3035.550000   
..                    ...                  ...                  ...   
902                2019.0                  0.0          2431.823590   
903                2019.0                  0.0          2431.823590   
904                2019.0                  0.0          2431.823590   
905                2019.0                  0.0          2431.823590   
906                2019.0                  0.0          2431.823590   

     generation_gwh_2015  generation_gwh_2016  generation_gwh_2017  \
0            2428.226946          2467.936859          2547.759305   
1            2428.226946          2467.936859          2547.759305   
2            2428.226946          2467.936859          2547.759305   
3             843.747000           886.004428           663.774500   
4            5916.370000          6243.000000          5385.579736   
..                   ...                  ...                  ...   
902             0.994875           233.596650           865.400000   
903          2428.226946          2467.936859          2547.759305   
904          2428.226946          2467.936859          2547.759305   
905          2428.226946          2467.936859          2547.759305   
906          2428.226946          2467.936859          2547.759305   

     generation_gwh_2018  generation_gwh_2019         generation_data_source  \
0            2600.804099                  0.0                              0   
1            2600.804099                  0.0                              0   
2            2600.804099                  0.0                              0   
3             626.239128                  0.0  Central Electricity Authority   
4            7279.000000                  0.0  Central Electricity Authority   
..                   ...                  ...                            ...   
902           686.50

### i define a list of columns to keep and use it to select the relevant columns for analysis effectively reducinng the dataset to only the columns i want to work with.

### I remove the 'Owner' columns as it appears  to have been selected for removal.

In [33]:
from sklearn.preprocessing import LabelEncoder

label_encoder = LabelEncoder()
df['other_fuel1'] = df['other_fuel1'].astype(str)
df['other_fuel2'] = df['other_fuel2'].astype(str)
df['other_fuel3'] = df['other_fuel3'].astype(str)
df['geolocation_source'] = df['geolocation_source'].astype(str)
df['primary_fuel'] = df['primary_fuel'].astype(str)
df['country'] = df['country'].astype(str)
df['country_long'] = df['country_long'].astype(str)
df['generation_data_source'] = df['generation_data_source'].astype(str)
# Apply label encoding
df['other_fuel1'] = label_encoder.fit_transform(df['other_fuel1'])
df['other_fuel2'] = label_encoder.fit_transform(df['other_fuel2'])
df['other_fuel3'] = label_encoder.fit_transform(df['other_fuel3'])
df['geolocation_source'] = label_encoder.fit_transform(df['geolocation_source'])
df['primary_fuel'] = label_encoder.fit_transform(df['primary_fuel'])
df['country'] = label_encoder.fit_transform(df['country'])
df['country_long'] = label_encoder.fit_transform(df['country_long'])
df['generation_data_source'] = label_encoder.fit_transform(df['generation_data_source'])


### I applied label encoding to certain categorical columns converting them to nemerical values this is done 
### using 'LabelEncoder' from sckit-learn

In [34]:
df

primary_fuel  country  capacity_mw  country_long  latitude  longitude  \
0               6        0          2.5             0   28.1839    73.2407   
1               1        0         98.0             0   24.7663    74.6090   
2               7        0         39.2             0   21.9038    69.3732   
3               2        0        135.0             0   23.8712    91.3602   
4               1        0       1800.0             0   21.9603    82.4091   
..            ...      ...          ...           ...       ...        ...   
902             1        0       1600.0             0   16.2949    77.3568   
903             6        0          3.0             0   12.8932    78.1654   
904             7        0         25.5             0   15.2758    75.5811   
905             1        0         80.0             0   24.3500    73.7477   
906             7        0         16.5             0    9.9344    77.4768   

     other_fuel1  other_fuel2  other_fuel3  commissioning_year  ...  \
0              0            0            0         2011.000000  ...   
1              0            0            0         1997.091082  ...   
2              0            0            0         1997.091082  ...   
3              0            0            0         2004.000000  ...   
4              3            0            0         2015.000000  ...   
..           ...          ...          ...                 ...  ...   
902            3            0            0         2016.000000  ...   
903            0            0            0         1997.091082  ...   
904            0            0            0         1997.091082  ...   
905            0            0            0         1997.091082  ...   
906            0            0            0         1997.091082  ...   

     year_of_capacity_data  generation_gwh_2013  generation_gwh_2014  \
0                   2019.0                  0.0          2431.823590   
1                   2019.0                  0.0          2431.823590   
2                   2019.0                  0.0          2431.823590   
3                   2019.0                  0.0           617.789264   
4                   2019.0                  0.0          3035.550000   
..                     ...                  ...                  ...   
902                 2019.0                  0.0          2431.823590   
903                 2019.0                  0.0          2431.823590   
904                 2019.0                  0.0          2431.823590   
905                 2019.0                  0.0          2431.823590   
906                 2019.0                  0.0          2431.823590   

     generation_gwh_2015  generation_gwh_2016  generation_gwh_2017  \
0            2428.226946          2467.936859          2547.759305   
1            2428.226946          2467.936859          2547.759305   
2            2428.226946          2467.936859          2547.759305   
3             843.747000           886.004428           663.774500   
4            5916.370000          6243.000000          5385.579736   
..                   ...                  ...                  ...   
902             0.994875           233.596650           865.400000   
903          2428.226946          2467.936859          2547.759305   
904          2428.226946          2467.936859          2547.759305   
905          2428.226946          2467.936859          2547.759305   
906          2428.226946          2467.936859          2547.759305   

     generation_gwh_2018  generation_gwh_2019  generation_data_source  \
0            2600.804099                  0.0                       0   
1            2600.804099                  0.0                       0   
2            2600.804099                  0.0                       0   
3             626.239128                  0.0                       1   
4            7279.000000                  0.0                       1   
..                   ...                  ...                     ...  

In [38]:
X = df.drop(['capacity_mw'], axis=1)
Y = df['capacity_mw']


In [39]:
from sklearn.model_selection import train_test_split
(X_train, X_test, y_train, y_test) = train_test_split(X,y,test_size=0.30,random_state=70)

In [40]:
from sklearn.linear_model import LinearRegression
model=LinearRegression()

In [41]:
X_train

primary_fuel  country  country_long  latitude  longitude  other_fuel1  \
631             1        0             0   24.9643    74.1197            0   
299             6        0             0   23.7162    87.0695            0   
749             7        0             0   10.8166    77.3199            0   
174             2        0             0   26.7024    77.8492            0   
551             6        0             0   25.3723    81.8717            0   
..            ...      ...           ...       ...        ...          ...   
316             3        0             0    8.9553    77.0697            0   
792             4        0             0   19.8294    72.6617            0   
114             6        0             0   27.3827    72.3188            0   
214             1        0             0   22.8618    78.8648            3   
334             1        0             0   24.7773    76.7370            3   

     other_fuel2  other_fuel3  commissioning_year  geolocation_source  ...  \
631            0            0         1997.091082                   3  ...   
299            0            0         1997.091082                   1  ...   
749            0            0         1997.091082                   3  ...   
174            0            0         2007.000000                   3  ...   
551            0            0         1997.091082                   1  ...   
..           ...          ...                 ...                 ...  ...   
316            0            0         1994.000000                   3  ...   
792            0            0         1996.000000                   3  ...   
114            0            0         1997.091082                   1  ...   
214            0            0         2018.000000                   3  ...   
334            0            0         2013.000000                   3  ...   

     year_of_capacity_data  generation_gwh_2013  generation_gwh_2014  \
631                 2019.0                  0.0           2431.82359   
299                 2019.0                  0.0           2431.82359   
749                 2019.0                  0.0           2431.82359   
174                 2019.0                  0.0            829.11420   
551                 2019.0                  0.0           2431.82359   
..                     ...                  ...                  ...   
316                 2019.0                  0.0           2431.82359   
792                 2019.0                  0.0           9335.00000   
114                 2019.0                  0.0           2431.82359   
214                 2019.0                  0.0           2431.82359   
334                 2019.0                  0.0           7399.00000   

     generation_gwh_2015  generation_gwh_2016  generation_gwh_2017  \
631          2428.226946          2467.936859          2547.759305   
299          2428.226946          2467.936859          2547.759305   
749          2428.226946          2467.936859          2547.759305   
174           282.700000            94.855000           210.230500   
551          2428.226946          2467.936859          2547.759305   
..                   ...                  ...                  ...   
316          2428.226946          2467.936859          2547.759305   
792          9453.000000          9887.200000          6173.620000   
114          2428.226946          2467.936859          2547.759305   
214          2428.226946          2467.936859          2547.759305   
334          8227.000000          7856.000000          4788.000000   

     generation_gwh_2018  generation_gwh_2019  generation_data_source  \
631          2600.804099                  0.0                       0   
299          2600.804099                  0.0                       0   
749          2600.804099                  0.0                       0   
174            20.234900                  0.0                       1   
551          2600.804099                  0.0               

In [43]:
y_train

631     160.0
299       2.0
749      21.0
174     330.0
551       5.8
        ...  
316      15.0
792    1400.0
114      15.0
214     800.0
334    1320.0
Name: capacity_mw, Length: 634, dtype: float64

In [44]:
model.fit(X_train,y_train)

LinearRegression()

In [45]:
model.predict(X_train)

array([ 7.04014183e+01, -7.03587543e+01,  4.40718309e+01,  2.53061089e+02,
       -3.64792274e+01,  4.84712874e+01,  3.34273429e+02,  5.08040734e+02,
        1.80173460e+02,  4.25304512e+01,  9.10922796e+01,  2.98796717e+01,
        1.36823657e+03,  3.27894426e+01,  1.02808943e+02, -1.04910543e+02,
        6.56312439e+01,  2.94910726e+01,  2.84535546e+03,  8.28346078e+02,
        8.40547618e+02,  5.54816262e+01,  7.29255455e+01,  7.38308280e+01,
        3.53414734e+02,  3.33734667e+01,  8.54205397e+01,  8.19862511e+02,
        2.25223372e+02, -2.13562511e+02,  1.10127556e+03,  8.49841300e+02,
        4.40930981e+01,  9.48044471e+01,  1.26335712e+02,  3.82271087e+02,
        6.06061290e+01,  1.53794768e+03,  1.46090653e+02,  6.60150668e+02,
        2.17371110e+03,  4.32999768e+02,  3.67850808e+02,  2.78227114e+02,
        1.52968410e+00,  2.00737885e+01,  6.56057107e+01,  2.52211338e+02,
       -1.48208752e+02,  1.75513232e+02,  4.10977664e+02,  1.67948574e+01,
        5.00251277e+01,  

In [46]:
model.score(X_train,y_train)

0.8471905827626194

In [47]:
from sklearn.model_selection import ShuffleSplit
from sklearn.model_selection import cross_val_score

cv = ShuffleSplit(n_splits=5, test_size=0.2, random_state=0)

cross_val_score(LinearRegression(), X, y, cv=cv)

array([0.91997447, 0.78553019, 0.74500913, 0.86350754, 0.86439958])

In [49]:
X = df.drop(['capacity_mw'], axis=1)
y = df['capacity_mw']

In [50]:
from sklearn.ensemble import RandomForestRegressor
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error
rf_regressor = RandomForestRegressor(n_estimators=100, random_state=42)
rf_regressor.fit(X_train, y_train)
rf_regressor.score(X_train, y_train)

0.9784193934312779

In [51]:
y_pred = rf_regressor.predict(X_test)
y_pred

array([3.79111355e+01, 3.12225000e+01, 6.41150000e+01, 1.43297600e+02,
       4.56350000e+01, 7.68180000e+01, 3.79111355e+01, 9.32590000e+01,
       4.72541200e+02, 2.31940000e+01, 3.35767000e+01, 2.62310000e+01,
       3.92030410e+02, 7.66000000e+00, 3.53878770e+02, 1.71643333e+01,
       3.80320000e+01, 1.65830000e+03, 1.51710000e+03, 1.16126500e+02,
       5.26100000e+00, 5.95971740e+02, 9.49150000e+01, 3.48268790e+02,
       2.25287733e+02, 8.17330000e+01, 3.79111355e+01, 1.33430000e+01,
       1.57975000e+01, 4.76323860e+02, 3.96568160e+02, 2.54390000e+01,
       1.51645000e+01, 1.24420000e+01, 4.96450550e+02, 4.73190000e+01,
       1.55352900e+02, 5.79548220e+02, 6.48307600e+02, 1.98910000e+03,
       7.70030000e+01, 1.26060700e+02, 1.10699160e+03, 1.45420000e+03,
       7.29248600e+02, 5.84203740e+02, 1.14222000e+03, 3.77260000e+01,
       4.31250000e+01, 1.08229000e+03, 1.15390000e+03, 2.12327080e+03,
       8.57870000e+01, 1.51470000e+01, 6.60172800e+02, 2.09470000e+01,
      

In [52]:
y_test

440      4.0
29     113.6
739    130.0
314    100.0
515     78.9
       ...  
97      69.6
646     48.0
55      20.0
480    101.0
801    420.0
Name: capacity_mw, Length: 273, dtype: float64

In [53]:
print('Training set score: {:.4f}'.format(rf_regressor.score(X_train, y_train)))

print('Test set score: {:.4f}'.format(rf_regressor.score(X_test, y_test)))

Training set score: 0.9784
Test set score: 0.8826


### The linear regression model has a training set score of approximately 0.8472. The cross validation results suggest some variability in model performance on different aplits of the data. the random forest regressor model perffors better with a training data. the test set score is approximately 0.88 suggesting that the model generalizes resonably well to new data.